### PR TITLE
Add quote search by game or show name

### PR DESCRIPTION
**Pull request by bunglelow**
Adds the ability to search for quotes by game or show name.
`!quote game Dark Souls` or `!quote show IDDQDerp`

Added as part of https://github.com/mrphlip/lrrbot/issues/165

### CODE REVIEW

File: lrrbot/commands/quote.py [hunk 1]
```diff
@@ -39,13 +39,14 @@ def format_quote(tag, qid, quote, name, date, context):
 		quote_msg += " [{date!s}]".format(date=date)
 	return quote_msg
 
-@bot.command("quote(?: (?:(\d+)|(.+)))?")
+@bot.command("quote\s*(?:(game|show)\s+(.+)|(?:(\d+)|(.+)))?")
```
Comment (andreasots) [inline lrrbot/commands/quote.py]: Spaces in the regular expression get replaced with `(?:\s+)`. So `'\s*'` should be `' ?'` and `'\s+'` should be `' '`. This regex also allows `!quotegame Dark Souls`. Is this intended?

File: lrrbot/commands/quote.py [hunk 2]
```diff
@@ -39,13 +39,14 @@ def format_quote(tag, qid, quote, name, date, context):
...
 @lrrbot.decorators.throttle(60, count=2)
-def quote(lrrbot, conn, event, respond_to, qid, attrib):
+def quote(lrrbot, conn, event, respond_to, meta_param, meta_value, qid, attrib):
 	"""
 	Command: !quote
 	Command: !quote ATTRIB
+	Command: !quote game|show VALUE
 	Section: quotes
 
 	Post a randomly selected quotation, optionally filtered by attribution.
```
Comment (andreasots) [inline lrrbot/commands/quote.py]: The command description isn't changed. It should explain what `!quote game|show VALUE` does.

**Comment by bunglelow** [comment]
Adding another set of brackets look messy to me initially, but my initial regex did mean you could ignore spaces between the command and the attributes. 

I've just pushed an update to the regex and command description.

File: lrrbot/commands/quote.py [hunk 3]
```diff
@@ -39,16 +39,17 @@ def format_quote(tag, qid, quote, name, date, context):
...
-@bot.command("quote(?: (?:(\d+)|(.+)))?")
+@bot.command("quote(?: (?:(game|show) (.+)|(?:(\d+)|(.+))))?")
 @lrrbot.decorators.sub_only
 @lrrbot.decorators.throttle(60, count=2)
-def quote(lrrbot, conn, event, respond_to, qid, attrib):
+def quote(lrrbot, conn, event, respond_to, meta_param, meta_value, qid, attrib):
 	"""
 	Command: !quote
 	Command: !quote ATTRIB
+	Command: !quote game|show VALUE
```
Comment (andreasots) [inline lrrbot/commands/quote.py]: This should be two entries `!quote game GAME` and `!game show SHOW` to match how other commands are described.

**Comment by bunglelow** [comment]
Updated

**Comment by andreasots** [comment]
Looks good, merging.

**Pull request merged**